From: Jeff Mahoney <jeffm@suse.com>
Subject: overlayfs: compat, fix incorrect dentry use in ovl_rename2
References: bsc#1032400
Patch-mainline: never, depends on overlayfs compat mode

ovl_rename2 also passes an upper dentry to ovl_is_whiteout, which causes
a NULL pointer deref when resolving the workdir using another file system's
fs_info.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/overlayfs/dir.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/overlayfs/dir.c
+++ b/fs/overlayfs/dir.c
@@ -885,7 +885,7 @@ static int ovl_rename2(struct inode *old
 	} else {
 		new_create = true;
 		if (!d_is_negative(newdentry) &&
-		    (!new_opaque || !ovl_is_whiteout(newdentry, new_upperdir)))
+		    (!new_opaque || !ovl_is_whiteout(newdentry, new)))
 			goto out_dput;
 	}
 
